python-lib/nlp_utils.py

```diff
@@ -3,16 +3,37 @@
 import unicodedata
 
 
-def get_keyword(text: AnyStr, normalize_case: bool) -> AnyStr:
+def get_text_case(text: AnyStr, normalize_case: bool) -> AnyStr:
     """Return text in its wanted-case form"""
     return text.lower() if normalize_case else text
 
+def get_attribute(lemmatize: bool) -> AnyStr:
+    """Return the right attribute to pass to spaCy Matcher"""
+    return "LEMMA" if lemmatize else "TEXT"
 
-def get_sentence(span: Span, normalize_case: bool) -> Union[Span, Doc]:
-    """Return Span object as a Doc if case_insensitive is set to True"""
-    return span if normalize_case else span.as_doc()
+
+def get_text_normalized(text: Doc, normalize_case: bool) -> AnyStr:
```

Comment:
Ditto, try and find a clearer function name

Reply:
0765d3c

Reply:
Nitpick: contrary to other functions in the nlp_utils module, this function operates on spacy Span objects. I believe that is an important difference, which is not obvious. I had to re-read the module multiple times to notice it.

As a convention, could functions that operate on strings be called `do_stuff_text`, those that operate on spacy tokens be called `do_stuff_token` and for spacy spans `do_stuff_span`?

Reply:
d807407